File: virtual-library/api/VirtualLibrary.Api.Tests/Domain/BookTests.cs
using FluentAssertions;
using VirtualLibrary.Api.Domain;
using Xunit;

namespace VirtualLibrary.Api.Tests.Domain;

public class BookTests
{
    [Fact]
    public void Book_Initialization_SetsProperties()
    {
        // Arrange & Act
        var book = new Book
        {
            Isbn = "9780134685991",
            Title = "Effective Java",
            Authors = new List<string> { "Joshua Bloch" },
            Publisher = "Addison-Wesley",
            PublishYear = 2018
        };

        // Assert
        book.Isbn.Should().Be("9780134685991");
        book.Title.Should().Be("Effective Java");
        book.Authors.Should().Contain("Joshua Bloch");
        book.Publisher.Should().Be("Addison-Wesley");
        book.PublishYear.Should().Be(2018);
    }

    [Theory]
    [InlineData("9780134685991")]
    [InlineData("978-0-13-468599-1")]
    public void Book_IsbnVariations_AreValid(string isbn)
    {
        // Act
        var book = new Book { Isbn = isbn, Title = "Test" };

        // Assert
        book.Isbn.Should().NotBeNullOrEmpty();
    }

    [Fact]
    public void Book_Authors_CanBeEmpty()
    {
        // Act
        var book = new Book { Title = "Test Book", Isbn = "123" };

        // Assert
        book.Authors.Should().BeEmpty();
    }
}

File: virtual-library/ios/VirtualLibraryTests/Models/BookTests.swift
import XCTest
@testable import VirtualLibrary

final class BookTests: XCTestCase {
    
    func testBookInitialization_WithAllProperties() {
        // Arrange
        let isbn = "9780134685991"
        let title = "Effective Java"
        let author = "Joshua Bloch"
        
        // Act
        let book = Book(
            isbn: isbn,
            title: title,
            author: author,
            publisher: "Addison-Wesley",
            publishedDate: "2018",
            description: "Test description",
            pageCount: 416,
            categories: ["Programming"],
            thumbnail: "https://example.com/cover.jpg",
            language: "en"
        )
        
        // Assert
        XCTAssertEqual(book.isbn, isbn)
        XCTAssertEqual(book.title, title)
        XCTAssertEqual(book.author, author)
        XCTAssertEqual(book.pageCount, 416)
    }
    
    func testBookEquality() {
        // Arrange
        let book1 = Book(isbn: "123", title: "Book", author: "Author")
        let book2 = Book(isbn: "123", title: "Book", author: "Author")
        let book3 = Book(isbn: "456", title: "Book", author: "Author")
        
        // Assert
        XCTAssertEqual(book1.isbn, book2.isbn)
        XCTAssertNotEqual(book1.isbn, book3.isbn)
    }
}
